### PR TITLE
Issue #151 Add codefix for S2933 (field should be readonly)

### LINKED ISSUE #151: Add codefix for S2933 (field should be readonly)

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #151

### CODE REVIEW

**Comment by tvsonar** [comment]
From the SyntaxTree, we can get all the diagnostics, and then filter them by ID, check for common field declarations, and then we have all the issues belonging to a single declaration.

File: src/SonarLint/Rules/FieldShouldBeReadonlyCodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        Title),
+                    context.Diagnostics);
+            }
+        }
+
+        private static async Task<Document> AddReadonlyKeyword(Document document, FieldDeclarationSyntax fieldDeclaration,
+            CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken);
+            var newFieldDeclaration = fieldDeclaration.AddModifiers(
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/FieldShouldBeReadonlyCodeFixProvider.cs]: @tvsonar this is quite cool, it adds it automatically in the usual order

File: src/SonarLint/Rules/FieldShouldBeReadonlyCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                if (fieldDeclaration == null)
+                {
+                    return;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        Title,
+                        c => AddReadonlyKeyword(context.Document, fieldDeclaration, c),
+                        Title),
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/FieldShouldBeReadonlyCodeFixProvider.cs]: @tvsonar as discussed - let's remove this

**Comment by dbolkensteyn** [comment]
LGTM - but the analyzer could IMO also suggest to make `private static` fields `readonly` - I've created a new issue for that: https://github.com/SonarSource/sonarlint-vs/issues/206

**Pull request merged**